### ISSUE TITLE
Auto-generated FK names don't match MySQL for renamed tables

### ISSUE THREAD

**Issue by arvidfm** [dolthub/dolt#7959]
In MySQL, if you rename a table, any autogenerated foreign keys will be renamed as well to match the new table name. This is not the case with Dolt, making it harder to write migrations operating on previously renamed tables that work with both MySQL and Dolt.

MWE, works with Percona Server 8.3, but fails with Dolt at `DROP FOREIGN KEY` because the FK is still named `b_test_ibfk_1` instead of `c_test_ibfk_1`.
```sql
DROP TABLE IF EXISTS c_test, b_test, a_test;

CREATE TABLE a_test (id INT PRIMARY KEY);

CREATE TABLE b_test(
    a_id INT,
    FOREIGN KEY (a_id) REFERENCES a_test (id)
);

RENAME TABLE b_test TO c_test;

ALTER TABLE c_test DROP FOREIGN KEY c_test_ibfk_1;

DROP TABLE IF EXISTS c_test, b_test, a_test;
```

**Comment by fulghum** [comment]
Thanks for reporting this difference with MySQL's behavior. I'm starting to dig into this one now.

**Comment by fulghum** [comment]
@arvidfm – the fix for this one is merged into Dolt now, and will go out in our next release. We'll try to get a few more fixes checked in for you and run a release later tonight or tomorrow.

**Comment by fulghum** [comment]
@arvidfm – The fix for this issue has been released in [Dolt 1.39.4](https://github.com/dolthub/dolt/releases/tag/v1.39.4). Thanks again for pointing out this behavior difference so we could fix it! 🙏  Let us know if you find anything else.
